### ISSUE TITLE
Remove the q map

### ISSUE THREAD

**Issue by odnoletkov** [tpope/vim-fugitive#1221]
It is very intrusive. I keep accidentally closing `:Gstatus` (and often Vim since I often `$ vim .git/index` these days) when trying to <kbd>q:?</kbd> for reverse commands search – which I do very often in this context to recall long `:Git` commands.
For `:Gblame` I don't care so much but it provides no value over <kbd>CTRL-W q</kbd>. I see the usefulness of `:bdelete` over <kbd>CTRL-W q</kbd> when `:Gstatus` is the last window, but that seems like a general not fugitive's problem. Am I missing something else? Alternatively it can be remapped to <kbd>gq</kbd>.

**Comment by tpope** [comment]
The status window comes and goes enough that I think shaving a couple of keystrokes off of closing it is not a ridiculous endeavor, but as the status window gets more functionality I increasingly agree it's a poor trade-off. For now, I will add a `gq` map. I think `q` probably has enough usage that it needs a deprecation cycle. After 3.0, feel free to send a PR that adds `<Bar>echohl WarningMsg<Bar>echo "q is deprecated in favor of gq"<Bar>echohl NONE` to the `q` map.

**Comment by SolaWing** [comment]
`q` is a commonly hot key to close window. And I rarely saw someone use `gq` to quit.
if `q` conflict with someone's habit, I suggest that allow user map or unmap it.
if have to use two-key-combo, I suggest use one of vim default key: `ZQ`, `CTRL-W q`, `CTRL-W CTRL-q`.

**Comment by sodapopcan** [comment]
TL;DR: I advocate for keeping `q`... mostly.

This is interesting as the command line window is something I avoid at all costs, especially with Vim's, uh, completion-based-on-what's-already-been-typed (I either forget or never know what this was called... but `:Git <Up>` lets you cycle through only history starting with `:Git `).  While I don't exactly pour through github issues in plugins I use on a regular basis, I do from time-to-time and this is the first I've seen of someone who actually uses the command line window... and, of course, that is totally legit.

That said, `q` for quit in unmodifiable output buffers is used by several respected plugin authors and I feel it's become a ubiquitous mapping for a whole slue of folk who use Vim a certain way.  Of course, this is hardly based on any real data, just my perception in small sample of plugins I use.

If you do decide to change, I do agree with @SolaWing that overriding a Vim default would be best.  The whole thing about using `g`-prefixed mappings is that `g` has become somewhat of my leader key, which I learned from Drew Neil who I believe learned it from you *winky-nudgey-stick-out-tongue emoji*

I actually don't hate `gq`, but I think consistency for muscle memory is important here and maybe wait for more complaints before committing to this change.

**Comment by tpope** [comment]
It is indeed common for read-only plugin buffers (which is why I did it), but overrides a built-in that has valid uses in read-only plugin buffers (which is I why I will ultimately remove it). `gq` is *slightly* easier to type than `<C-W>q` so it does offer a tiny bit of value on that front. I don't understand `g` as leader key, 20 of the 26 lowercase `g` followups (including q) are taken. It's low value enough that I'd be fine to check for another `gq` map and defer to it, if that's really a concern.

**Comment by justinmk** [comment]
`gq` is a good choice IMO, and I'm considering it for https://github.com/justinmk/vim-dirvish/ as well. `q`-to-quit never felt right to me, it often gets in the way when I want to end macro-recording.

**Comment by d3jn** [comment]
@tpope, since `q` map is deprecated as of now and later (as far as I understand) will be removed, can we have an option that enables this map and keeps its previous behavior for this plugin?

I am so used to `q` for popular plugins (such as NerdTree, VimMerginal and so on) that I include it for my own plugins as well.

**Comment by tpope** [comment]
See conversation at https://github.com/tpope/vim-fugitive/commit/a510b3aadf3f39711c113371c18adc48ad54e6ee#commitcomment-35424504.

**Comment by dyng** [comment]
For those who misses the old fashion `q`, add below line in your `.vimrc`

```
autocmd FileType fugitiveblame nmap <buffer> q gq
```

**Comment by alexventuraio** [comment]
I added this file in my Lua config with the following content but I got the same error:
```lua
-- ftplugin/fugitiveblame.lua
vim.keymap.set('n', 'q', '[[gq]]', { buffer = 0, nowait = true, noremap=true })
```
<img width="552" alt="image" src="https://github.com/tpope/vim-fugitive/assets/13169164/b8776cd0-6623-4e3e-889b-703686440ed2">

Did you happen to got that fixed? @bangedorrunt

**Comment by kdowolski** [comment]
@bangedorrunt  @alexventuraio 
The problem is `gq` is already a mapping, so your mapping should be recursive (delete `noremap`)

My version:
```
vim.api.nvim_create_autocmd("FileType", {
    pattern = {"fugitiveblame", "fugitive"},
    callback = function()
        vim.api.nvim_buf_set_keymap(0, 'n', 'q', 'gq', { silent = true })
    end
})
```
